contracts/src/ERC20Permit.sol

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.20;
+
+import {IERC20} from "./interfaces/IERC20.sol";
+import {IERC20Permit} from "./interfaces/IERC20Permit.sol";
+
+import {ERC20} from "./ERC20.sol";
+import {Ownable} from "openzeppelin/access/Ownable.sol";
```

Comment:
I don't really want to rely on Openzeppelin. I've removed most of its use from the codebase. See my preferred solution in the comments below.